checkout/interfaces/controller/apicontroller.go

```diff
@@ -0,0 +1,244 @@
+// @Router /api/v1/checkout/placeorder [put]
+func (c *APIController) StartPlaceOrderAction(ctx context.Context, r *web.Request) web.Result {
+	session := web.SessionFromContext(ctx)
+	cart, err := c.cartService.GetCartReceiverService().ViewCart(ctx, session)
+	if err != nil {
+		response := c.responder.Data(errorResponse{Code: "500", Message: err.Error()})
+		response.Status(http.StatusInternalServerError)
+		return response
+	}
+	returnURLRaw, err := r.Query1("returnURL")
+	if err != nil {
+		response := c.responder.Data(errorResponse{Code: "400", Message: "returnURL missing"})
+		response.Status(http.StatusBadRequest)
+		return response
+	}
+	var returnURL *url.URL
+	returnURL, err = url.Parse(returnURLRaw)
+	if err != nil {
+		response := c.responder.Data(errorResponse{Code: "400", Message: err.Error()})
+		response.Status(http.StatusBadRequest)
+		return response
+	}
+
+	startPlaceOrderCommand := placeorder.StartPlaceOrderCommand{Cart: *cart, ReturnURL: returnURL}
+	pctx, err := c.placeorderHandler.StartPlaceOrder(ctx, startPlaceOrderCommand)
+	if err == placeorder.ErrAnotherPlaceOrderProcessRunning {
+		dtopctx, err := c.placeorderHandler.RefreshPlaceOrder(ctx, placeorder.RefreshPlaceOrderCommand{})
+		if err != nil {
+			response := c.responder.Data(errorResponse{Code: "500", Message: err.Error()})
+			response.Status(http.StatusInternalServerError)
+			return response
+		}
+		return c.responder.Data(startPlaceOrderResult{
+			UUID: dtopctx.UUID,
+		})
+
+	}
+	if err != nil {
+		response := c.responder.Data(errorResponse{Code: "500", Message: err.Error()})
+		response.Status(http.StatusInternalServerError)
+		return response
+	}
+	response := c.responder.Data(startPlaceOrderResult{
+		UUID: pctx.UUID,
+	})
+	response.Status(http.StatusCreated)
+	return response
+}
+
+// CancelPlaceOrderAction cancels a running place order process
+// @Summary Cancels a running place order process
+// @Tags v1 Checkout ajax API
+// @Produce json
+// @Success 200 {boolean} boolean
+// @Failure 500 {object} errorResponse
+// @Router /api/v1/checkout/placeorder/cancel [post]
+func (c *APIController) CancelPlaceOrderAction(ctx context.Context, r *web.Request) web.Result {
+	err := c.placeorderHandler.CancelPlaceOrder(ctx, placeorder.CancelPlaceOrderCommand{})
+	if err != nil {
+		response := c.responder.Data(errorResponse{Code: "500", Message: err.Error()})
+		response.Status(http.StatusInternalServerError)
+		return response
+	}
+	return c.responder.Data(true)
+}
+
+// ClearPlaceOrderAction clears the last place order if in final state
+// @Summary Clears the last placed order if in final state
+// @Tags v1 Checkout ajax API
+// @Produce json
+// @Success 200 {boolean} boolean
+// @Failure 500 {object} errorResponse
+// @Router /api/v1/checkout/placeorder [delete]
+func (c *APIController) ClearPlaceOrderAction(ctx context.Context, r *web.Request) web.Result {
+	err := c.placeorderHandler.ClearPlaceOrder(ctx)
+	if err != nil {
```

Comment:
should we use another status code whenever there is no place order process running? in this case we currently return `{"Code":"500","Message":"ErrNoPlaceOrderProcess"}`